OrbitVulkanLayer/include/OrbitVulkanLayer/CommandBufferManager.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_LAYER_COMMAND_BUFFER_MANAGER_H_
+#define ORBIT_LAYER_COMMAND_BUFFER_MANAGER_H_
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "absl/synchronization/mutex.h"
+#include "vulkan/vulkan.h"
+
+namespace orbit::layer {
+
+/*
+ * This class is responsible to track command buffers and command pools.
+ * TODO: So far it only tracks the allocation/de-allocation of buffers and pools.
+ *  It should probably also track the timestamps inside the buffers (and the markers?)
+ *
+ * It also tracks which command buffer belongs to which device, which can be used
+ * in the `DispatchTable` for function look-up.
+ *
+ * Thread-Safety: This class is internally synchronized (using read/write locks), and can be
+ * safely accessed from different threads.
+ *
+ *
+ */
+class CommandBufferManager {
+ public:
+  void TrackCommandPool(VkCommandPool pool);
+  void UnTrackCommandPool(VkCommandPool pool);
+  void TrackCommandBuffers(VkDevice device, VkCommandPool pool,
+                           const VkCommandBuffer* command_buffers, uint32_t count);
+  void UnTrackCommandBuffers(VkDevice device, VkCommandPool pool,
+                             const VkCommandBuffer* command_buffers, uint32_t count);
+
+  [[nodiscard]] bool IsCommandPoolTracked(const VkCommandPool& pool);
```

Comment:
Should method be "const" (and mutex be mutable)?

Same for the next two methods.